File: Genogram.Client/src/app/core/models/Relationship.ts
import { Child } from "./Child";

export interface Relationship {
    id: number;
    childId: number;
    child: Child; 
    firstName: string;
    lastName?: string; 
    relationshipType: string;
    email?: string; 
    phoneNumber?: number; 
    isPrimaryContact: boolean;
    remarks?:string;
    actions?:string;
  }

File: Genogram.Domain/Entities/Relationship.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Genogram.Domain.Entities
{
    public class Relationship
    {
        public int Id { get; set; }
        public int ChildId { get; set; }
        public Child Child { get; set; } = null!;
        public string FirstName { get; set; } = null!;
        public string? LastName { get; set; }
        public string RelationshipType { get; set; } = null!;
        public string? Email { get; set; } 
        public string? PhoneNumber { get; set; } 
        public bool IsPrimaryContact { get; set; }
        public string? Remarks { get; set; }
    }
}
